python/lsst/pipe/tasks/processBrightStars.py

```diff
@@ -0,0 +1,414 @@
+            "lanczos3": "Lanczos kernel of order 3",
+            "lanczos4": "Lanczos kernel of order 4",
+            "lanczos5": "Lanczos kernel of order 5",
+        }
+    )
+    annularFluxRadii = pexConfig.ListField(
+        dtype=int,
+        doc="Inner and outer radii of the annulus used to compute the AnnularFlux for normalization, "
+            "in pixels.",
+        default=(40, 50)
+    )
+    annularFluxStatistic = pexConfig.ChoiceField(
+        dtype=str,
+        doc="Type of statistic to use to compute annular flux.",
+        default="MEANCLIP",
+        allowed={
+            "MEAN": "mean",
+            "MEDIAN": "median",
+            "MEANCLIP": "clipped mean",
+        }
+    )
+    numSigmaClip = pexConfig.Field(
+        dtype=float,
+        doc="Sigma for outlier rejection; ignored if annularFluxStatistic != 'MEANCLIP'.",
+        default=4
+    )
+    numIter = pexConfig.Field(
+        dtype=int,
+        doc="Number of iterations of outlier rejection; ignored if annularFluxStatistic != 'MEANCLIP'.",
+        default=3
+    )
+    badMaskPlanes = pexConfig.ListField(
+        dtype=str,
+        doc="Mask planes that, if set, lead to associated pixels not being included in the computation of the"
+            " annular flux.",
+        default=('BAD', 'CR', 'CROSSTALK', 'EDGE', 'NO_DATA', 'SAT', 'SUSPECT', 'UNMASKEDNAN')
+    )
+    refCatLoader = pexConfig.ConfigurableField(
+        target=LoadIndexedReferenceObjectsTask,
+        doc="reference object loader for astrometric calibration",
+    )
+
+    def setDefaults(self):
+        self.refCatLoader.ref_dataset_name = "gaia_dr2_20200414"
+
+
+class ProcessBrightStarsTask(pipeBase.PipelineTask, pipeBase.CmdLineTask):
+    """Task to extract, process, and store small image cut-outs (or "postage
+    stamps") around bright stars.
+
+    `ProcessBrightStarsTask` relies on three methods, called in succession:
+
+    `extractStamps`
+        Find bright stars within the exposure using a reference catalog and
+        extract a stamp centered on each.
+    `warpStamps`
+        Shift and warp each stamp to remove optical distortions and sample all
+        stars on the same pixel grid.
+    `computeAnnularFlux`
+        Computes the flux of an object in an annulus. This is required to
+        normalize each bright star stamp as their central pixels are likely
+        saturated and/or contain ghosts, and cannot be used.
+    """
+    ConfigClass = ProcessBrightStarsConfig
+    _DefaultName = "processBrightStars"
+    RunnerClass = pipeBase.ButlerInitializedTaskRunner
+
+    def __init__(self, butler=None, initInputs=None, *args, **kwargs):
+        pipeBase.Task.__init__(self, *args, **kwargs)
+        # Compute (model) stamp size depending on provided "buffer" value
+        self.modelStampSize = (int(self.config.stampSize[0]*self.config.modelStampBuffer),
+                               int(self.config.stampSize[1]*self.config.modelStampBuffer))
```

Comment:
Two reasons to favor `int`:

- if the resulting stamp size corresponds to an even number of pixels, we will add an extra pixel. So we end up with either the integer number immediately below or immediately above the exact (real number) `stampSize*modelStampBuffer`. Going with `ceil` would lead to either the integer immediately higher, or the one above _that_.
- we will be saving quite a lot of stamps of that size, so since it hardly matters in terms of the resulting processed stamps (by construction, we are taking a buffer to ensure we don't lose any useful pixel even after warping), I figure it is always best to go with one less pixel in each dimension.